core/src/main/java/org/apache/brooklyn/util/core/flags/TypeCoercions.java

```diff
@@ -138,6 +139,9 @@ private TypeCoercions() {}
     @SuppressWarnings({ "unchecked" })
     public static <T> T coerce(Object value, TypeToken<T> targetTypeToken) {
         if (value==null) return null;
+        if (value instanceof DeferredSupplier<?>) {
+            value = ((DeferredSupplier<?>) value).get();
```

Comment:
@aledsage: Now that the coersion methods have been moved into `brooklyn-utils-common`, we can't implement this check as-is without adding a new dependency. `DeferredSupplier` is arguably also a "common utility" so could be moved, but that ovbiously has implications beyong this PR. Let's review.